circleci: failed tests: TestTxnDeadline

The following test appears to have failed:

[#19453](https://circleci.com/gh/cockroachdb/cockroach/19453):

```
I160622 16:03:34.623142 client/db.go:474  failed batch: read at time 1466615914.612288121,100 encountered previous write with future timestamp 1466615914.612288121,100 within uncertainty interval
I160622 16:03:34.623455 kv/dist_sender.go:983  application error: storage/store.go:1743: rejecting command with timestamp in the future: 1466615914612288121 (1h14m59.98893748s ahead)
W160622 16:03:34.623515 kv/dist_sender.go:728  failed to invoke Get [/Table/2/1/50/"test"/3/1,/Min): storage/store.go:1743: rejecting command with timestamp in the future: 1466615914612288121 (1h14m59.98893748s ahead)
I160622 16:03:34.623602 client/db.go:474  failed batch: storage/store.go:1743: rejecting command with timestamp in the future: 1466615914612288121 (1h14m59.98893748s ahead)
I160622 16:03:34.624326 storage/engine/rocksdb.go:186  closing in-memory rocksdb instance
--- FAIL: TestTxnDeadline (0.24s)
	txn_restart_test.go:942: pq: storage/store.go:1743: rejecting command with timestamp in the future: 1466615914612288121 (1h14m59.98893748s ahead)
FAIL
FAIL	github.com/cockroachdb/cockroach/sql	54.986s
I160622 16:02:45.166957 rand.go:76  Random seed: 1232083257087864769
=== RUN   TestProcessExpression
--- PASS: TestProcessExpression (0.00s)
=== RUN   TestOrderedSync
--- PASS: TestOrderedSync (0.00s)
=== RUN   TestJoinReader
I160622 16:02:45.183075 storage/engine/rocksdb.go:143  opening in memory rocksdb instance
W160622 16:02:45.189154 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160622 16:02:45.211572 server/node.go:401  store store=0:0 ([]=) not bootstrapped
W160622 16:02:45.212016 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node

```
Please assign, take a look and update the issue accordingly.

--- a/sql/txn_restart_test.go
+++ b/sql/txn_restart_test.go
@@ -868,17 +868,16 @@ func TestNonRetryableErrorFromCommit(t *testing.T) {
 // the clock, so that the transaction timestamp exceeds the deadline of the EndTransactionRequest.
 func TestTxnDeadline(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	t.Skip("#7394")
 
 	var cmdFilters CommandFilters
 	cmdFilters.AppendFilter(checkEndTransactionTrigger, true)
 
-	restartDone := false
+	clockUpdate := false
 	testKey := []byte("test_key")
 	testingKnobs := &storage.StoreTestingKnobs{
 		TestingCommandFilter: cmdFilters.runFilters,
 		ClockBeforeSend: func(c *hlc.Clock, ba roachpb.BatchRequest) {
-			if restartDone {
+			if clockUpdate {
 				return
 			}
 
@@ -886,6 +885,7 @@ func TestTxnDeadline(t *testing.T) {
 			for _, union := range ba.Requests {
 				if req, ok := union.GetInner().(*roachpb.ScanRequest); ok {
 					if bytes.Contains(req.Key, testKey) {
+						clockUpdate = true
 						now := c.Now()
 						now.WallTime += int64(5 * sql.LeaseDuration)
 						c.Update(now)
@@ -901,6 +901,7 @@ func TestTxnDeadline(t *testing.T) {
 	server, sqlDB, _ := setupWithContext(t, &ctx)
 	defer cleanup(server, sqlDB)
 
+	restartDone := false
 	cleanupFilter := cmdFilters.AppendFilter(
 		func(args storagebase.FilterArgs) *roachpb.Error {
 			if restartDone {

